Type error when retrieving a datetime value from database

The following error was found while retrieving a datetime value from the database
`panic: expected value to have type dbtype.Time but found type time.Time`

**Neo4j Version:** 5.7.0 Community
**Neo4j Mode**: Single instance 
**Driver version**: Go driver v5.8.1
**Go version**: go1.20.4 linux/amd64
**Operating System**: Linux Mint 21.1 Vera 64-bit

### Steps to reproduce
1. Start Neo4j locally
2. Create a node with datetime value
`create (n:Node{dateTimeLabel: datetime("2016-11-22T18:59:00.000+0900")});`
3. Attempting to retrieve node property triggers an error
```
result, err := neo4j.ExecuteQuery(ctx, driver, "MATCH (n:Node) RETURN n", map[string]any{}, neo4j.EagerResultTransformer)
if err != nil {
	panic(err)
}
node, _, err := neo4j.GetRecordValue[neo4j.Node](result.Records[0], "n")
if err != nil {
	panic(err)
}
dateTimeVal, err := neo4j.GetProperty[neo4j.Time](node, "dateTimeLabel")
if err != nil { //Non-nil error
	panic(err)
}
```
As an alternative, when `neo4j.Time` is replaced with `time.Time`, it does not match the type specification and an error is returned.
```
dateTimeVal, err := neo4j.GetProperty[time.Time](node, "dateTimeLabel")
if err != nil { //Non-nil error
	panic(err)
}
```
### Expected behavior
Perhaps the node property `dateTimeLabel` should be of type `dbtype.Time` to avoid returning the error?
### Actual behavior
Node property `dateTimeLabel` is of type `time.Time`
### Stacktrace
For
```
dateTimeVal, err := neo4j.GetProperty[neo4j.Time](node, "dateTimeLabel")
if err != nil { //Non-nil error
	panic(err)
}
```
```
panic: expected value to have type dbtype.Time but found type time.Time

goroutine 1 [running]:
main.main()
	/home/---------
exit status 2
```
For
```
dateTimeVal, err := neo4j.GetProperty[time.Time](node, "dateTimeLabel")
if err != nil { //Non-nil error
	panic(err)
}
```
```
./neo4j.go:131:38: time.Time does not satisfy neo4j.PropertyValue (time.Time missing in bool | int64 | float64 | string | github.com/neo4j/neo4j-go-driver/v5/neo4j/dbtype.Point2D | github.com/neo4j/neo4j-go-driver/v5/neo4j/dbtype.Point3D | github.com/neo4j/neo4j-go-driver/v5/neo4j/dbtype.Date | github.com/neo4j/neo4j-go-driver/v5/neo4j/dbtype.LocalTime | github.com/neo4j/neo4j-go-driver/v5/neo4j/dbtype.LocalDateTime | github.com/neo4j/neo4j-go-driver/v5/neo4j/dbtype.Time | github.com/neo4j/neo4j-go-driver/v5/neo4j/dbtype.Duration | []byte | []any
```
### Quick dirty fix
added `time.Time` to `type PropertyValue interface` in neo4j/graph.go

--- a/neo4j/graph.go
+++ b/neo4j/graph.go
@@ -21,12 +21,13 @@ package neo4j
 
 import (
 	"fmt"
+	"time"
 )
 
 type PropertyValue interface {
 	bool | int64 | float64 | string |
 		Point2D | Point3D |
-		Date | LocalTime | LocalDateTime | Time | Duration | /* OffsetTime == Time == dbtype.Time */
+		Date | LocalTime | LocalDateTime | Time | Duration | time.Time | /* OffsetTime == Time == dbtype.Time */
 		[]byte | []any
 }
 

--- a/neo4j/graph_test.go
+++ b/neo4j/graph_test.go
@@ -169,6 +169,15 @@ func TestGetProperty(outer *testing.T) {
 				AssertNoError(t, err)
 			})
 
+			inner.Run("datetimes", func(t *testing.T) {
+				entity := test.make(singleProp("k", now))
+
+				prop, err := neo4j.GetProperty[time.Time](entity, "k")
+
+				AssertDeepEquals(t, prop, now)
+				AssertNoError(t, err)
+			})
+
 			inner.Run("durations", func(t *testing.T) {
 				entity := test.make(singleProp("k", neo4j.DurationOf(5, 4, 3, 2)))
 

--- a/neo4j/record.go
+++ b/neo4j/record.go
@@ -19,12 +19,15 @@
 
 package neo4j
 
-import "fmt"
+import (
+	"fmt"
+	"time"
+)
 
 type RecordValue interface {
 	bool | int64 | float64 | string |
 		Point2D | Point3D |
-		Date | LocalTime | LocalDateTime | Time | Duration | /* OffsetTime == Time == dbtype.Time */
+		Date | LocalTime | LocalDateTime | Time | Duration | time.Time | /* OffsetTime == Time == dbtype.Time */
 		[]byte | []any | map[string]any |
 		Node | Relationship | Path
 }

--- a/neo4j/record_test.go
+++ b/neo4j/record_test.go
@@ -168,6 +168,16 @@ func TestGetRecordValue(outer *testing.T) {
 			AssertNoError(t, err)
 		})
 
+		inner.Run("datetimes", func(t *testing.T) {
+			entity := record("k", now)
+
+			value, isNil, err := neo4j.GetRecordValue[time.Time](entity, "k")
+
+			AssertDeepEquals(t, value, now)
+			AssertFalse(t, isNil)
+			AssertNoError(t, err)
+		})
+
 		inner.Run("durations", func(t *testing.T) {
 			entity := record("k", neo4j.DurationOf(5, 4, 3, 2))
 

